### PR TITLE
feat: UI Unit screenshot

### PR DESCRIPTION
**Pull request by caalador**
Documentation for getting a "screenshot"
when a UI Unit Test fails.

part of vaadin/testbench#1336

### CODE REVIEW

File: articles/tools/testbench/ui-unit-testing-screenshot.asciidoc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    │   │   │               ├── LineAwesomeIcon[@class='menu-item-icon la la-file']
+    │   │   │               └── Span[text='About', @class='menu-item-text']
+    │   │   └── Footer[@class='footer']
+    │   └── HelloWorldView[@theme='margin spacing']
+    │       ├── TextField[label='Your name', value='', @style='align-self:flex-end']
+    │       └── Button[caption='Say hello', @style='align-self:flex-end']
+    └── Notification[duration='5000', opened='true', position='bottom-start']
+----
+
+The feature is by default not used and can be enabled by adding `@ExtendWith(TreeOnFailureExtension.class)` to the test class in JUnit5.
```
Comment (mcollovati) [inline articles/tools/testbench/ui-unit-testing-screenshot.asciidoc:38]:
```suggestion
The feature is not active by default and can be enabled by adding `@ExtendWith(TreeOnFailureExtension.class)` to the test class in JUnit5.
```

File: articles/tools/testbench/ui-unit-testing-screenshot.asciidoc [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+For JUnit4 the `UIUnit4Test` contains the method `printTree` that by default returns false, so overriding this in the test and returning true will enable the feature for JUnit4.
```
Comment (mcollovati) [inline articles/tools/testbench/ui-unit-testing-screenshot.asciidoc:40]:
```suggestion
For JUnit4 tests based on [classname]`UIUnit4Test`, the feature can be activated by overriding the [methodname]`printTree()` method to return true.
```

File: articles/tools/testbench/ui-unit-testing-screenshot.asciidoc [hunk 3]
```diff
@@ -0,0 +1,40 @@
+---
+title: Screenshots in UI Unit Testing
+order: 161
+layout: page
+---
+
+[[testbench.uiunit.screenshot]]
+= Screenshots in UI Unit Testing
+
+UI Unit testing supports returning a text representation of the UI for failing tests.
```
Comment (dougjewsbury) [inline articles/tools/testbench/ui-unit-testing-screenshot.asciidoc:10]:
```suggestion
UI Unit testing supports returning a text representation of the UI for tests that fail.
```

File: articles/tools/testbench/ui-unit-testing-screenshot.asciidoc [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+.Screenshot Of Failing Test
```
Comment (dougjewsbury) [inline articles/tools/testbench/ui-unit-testing-screenshot.asciidoc:12]:
```suggestion
.Screenshot of Test That Fails
```

**Review by dougjewsbury — COMMENTED** [review]
AsciiDoc looks good.

**Comment by jouni** [comment]
FYI, only the “regular” page section headings need to use `Title Case`. Source code block and image titles should use `Sentence case`.

**Review by mcollovati — APPROVED** [review]

**Pull request merged**